### ISSUE TITLE
"300 degC" wrongly converted to "300 K"

### ISSUE THREAD

**Issue by jkotan** [SciCatProject/backend#926]
### Summary

"300 degC" is wrongly converted to "300 K". Probably it is a bug in mathjs. I tested with (10.2.4).

### Steps to Reproduce

to reproduce it one can store a scalar value with 'degC' unit or  simpler in nodejs
```js
 unit("300 degC").toSI().toJSON()


### Current Behaviour

```json
{ mathjs: 'Unit', value: 300, unit: 'K', fixPrefix: true }
```

### Expected Behaviour

```js
{ mathjs: 'Unit', value: 573.15, unit: 'K', fixPrefix: true }
```

### Extra Details

Workaround for the problem could be to change the above line of the code in convertToSI to something like
```js
 unit(inputValue, inputUnit).to(unit(inputUnit).toSI()).toJSON()
```

which gives the correct result
```js
{ mathjs: 'Unit', value: 573.15, unit: 'K', fixPrefix: true }
```

**Comment by jkotan** [comment]
I've added the patch to #925

**Comment by bpedersen2** [comment]
Related to https://github.com/josdejong/mathjs/issues/2499 ( and a mathjs upgrade on our side?)

**Comment by jkotan** [comment]
Hi @bpedersen2,

In the  mathjs 12.1.0 still exists so it looks like the problem was not fixed by  josdejong/mathjs#2499 
```js
(mynodeenv) (myenv) jkotan@xpsjk:~/nest2$ nodejs 
Welcome to Node.js v21.3.0.
Type ".help" for more information.
>  const { unit, version } = await import("mathjs");
undefined
> version
'12.1.0'
>  unit("300 degC").toSI().toJSON()
{ mathjs: 'Unit', value: 300, unit: 'K', fixPrefix: true }
>  unit(300, "degC").to(unit("degC").toSI()).toJSON()
{ mathjs: 'Unit', value: 573.15, unit: 'K', fixPrefix: true }
```

**Comment by nitrosx** [comment]
@jkotan given that the discussion regarding arrays in Scientific Metadata requires more time to be resolved, could you please isolate the code to solve the problem highlighted in this issue in a separate PR?
Thanks
